spec/lib/administrate/search_spec.rb

```diff
@@ -59,4 +63,297 @@ class User; end
+            class User
+              def self.with_argument(argument); argument; end
+            end
+            search = Administrate::Search.new(resolver,
+                                              "scope:#{scope_with_argument}")
+            expect(search.scope).to eq(scope)
+            expect(search.scopes).to eq([scope])
+            expect(search.arguments).to eq([argument])
+          ensure
+            remove_constants :User
+          end
+        end
+      end
+    end
+
+    describe "the query is a word and a scope" do
+      let(:word) { "foobar" }
+
+      it "returns the scope and #words the word" do
+        begin
+          class User
+            def self.active; end
+          end
+
+          search = Administrate::Search.new(resolver, "scope:#{scope} #{word}")
+          expect(search.scope).to eq(scope)
+          expect(search.words).to eq([word])
+        ensure
+          remove_constants :User
+        end
+      end
+
+      it "ignores the order" do
+        begin
+          class User
+            def self.active; end
+          end
+
+          search = Administrate::Search.new(resolver, "#{word} scope:#{scope}")
+          expect(search.scope).to eq(scope)
+          expect(search.words).to eq([word])
+        ensure
+          remove_constants :User
+        end
+      end
+    end
+
+    describe "the query is a word and two scopes" do
+      let(:word) { "foobar" }
+      let(:other_scope) { "subscribed" }
+
+      describe "in that order" do
+        let(:query) { "#{word} scope:#{scope} scope:#{other_scope}" }
+
+        it "returns the scopes and #words the word" do
+          begin
+            class User
+              def self.active; end
+
+              def self.subscribed; end
+            end
+ 
+            search = Administrate::Search.new(resolver, query)
+            expect(search.scopes).to eq([scope, other_scope])
+            expect(search.words).to eq([word])
+          ensure
+            remove_constants :User
+          end
+        end
+      end
+
+      describe "with the word between the two scopes" do
+        let(:query) { "scope:#{scope} #{word} scope:#{other_scope}" }
+
+        it "returns the scopes and #words the word" do
+          begin
+            class User
+              def self.active; end
+
+              def self.subscribed; end
+            end
+            search = Administrate::Search.new(resolver, query)
+
+            expect(search.scopes).to eq([scope, other_scope])
+            expect(search.words).to eq([word])
+          ensure
+            remove_constants :User
+          end
+        end
+      end
+    end
+
+    describe "the query is one scope with an argument" do
+      let(:scope) { "name_starts_with" }
+      let(:argument) { "A" }
+      let(:query) { "scope:#{scope}(#{argument})" }
+
+      it "returns the [scope] and #arguments the [argument]" do
+        begin
+          class User
+            def self.name_starts_with(letter); end
```

Comment:
Unused method argument - `letter`. If it's necessary, use `_` or `_letter` as an argument name to indicate that it won't be used. You can also write as `name_starts_with(*)` if you want the method to accept any arguments but don't care about them.